src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6073,4 +6073,25 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="IDS_FeatureModuleInitializers" xml:space="preserve">
     <value>module initializers</value>
   </data>
+  <data name="ERR_ModuleInitializerMethodMustBeAccessibleOutsideTopLevelType" xml:space="preserve">
+    <value>Module initializer method '{0}' must be accessible outside top-level type '{1}'</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustBeStatic" xml:space="preserve">
+    <value>Module initializer method '{0}' must be static</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotBeContainedInGenericType" xml:space="preserve">
+    <value>Module initializer method '{0}' must not be contained in a generic type</value>
+  </data>
+  <data name="ERR_ModuleInitializerMethodMustNotBeGeneric" xml:space="preserve">
+    <value>Module initializer method '{0}' must not have generic parameters</value>
```

Comment:
>generic parameters [](start = 57, length = 18)

"generic **type** parameters"?

Reply:
I thought generic type parameters were generic parameters on a type, and generic method parameters were generic parameters on a method. Is it valid to refer to a method's generic parameters as 'generic type parameters'?

Reply:
> I thought generic type parameters were generic parameters on a type, and generic method parameters were generic parameters on a method. Is it valid to refer to a method's generic parameters as 'generic type parameters'?

I think these are "type parameters", i.e. they are types, rather than on the type. 
```
  <Node Name="MethodDeclarationSyntax" Base="BaseMethodDeclarationSyntax">
       ...
       <Field Name="TypeParameterList" Type="TypeParameterListSyntax" Optional="true"/>
```

---
In reply to: [418176343](https://github.com/dotnet/roslyn/pull/43301#discussion_r418176343) [](ancestors = 418176343)

Reply:
Sure enough, I don't know what I'm talking about. Spec calls them 'type parameters.' I'll stop doing this.

Reply:
@AlekseyTs Which do you prefer:

> Module initializer method '{0}' must not declare generic type parameters

> Module initializer method '{0}' must not be generic

Reply:
> Sure enough, I don't know what I'm talking about. Spec calls them 'type parameters.' I'll stop doing this.

Sorry, your reply was not visible when I commented or there would have been no need.

Reply:
I would do by example:
```
  <data name="WRN_MainCantBeGeneric" xml:space="preserve">
    <value>'{0}': an entry point cannot be generic or in a generic type</value>
  </data>
```

---
In reply to: [418186519](https://github.com/dotnet/roslyn/pull/43301#discussion_r418186519) [](ancestors = 418186519)

Reply:
Specifically I refer to the "cannot be generic or in a generic type" part

---
In reply to: [418195368](https://github.com/dotnet/roslyn/pull/43301#discussion_r418195368) [](ancestors = 418195368,418186519)